Update bravia-tv backend

dcnielsen90 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
- Update bravia-tv to the latest release.
- Remove no longer required error handling.
- Removed mac address references -- bravia_tv dynamically handles this during connect() now.
- Fix and update the bravia entity update function. 
- Remove logic that requires TV to be on during HA reboot

The purpose of this PR is to address where home-assistant is rebooted and the BraviaTV is off, the BraviaTV instance will be a bad state where it is not controllable. This is a result of the BraviaTV instance held by home-assistant does not have the proper cookies and the RESTapi subtlety rejects all the commands. In addition to this, the logic that prevents home-assistant from loading a BraviaTV instance at boot needed to be removed. This PR also includes some code enhancements in the areas that were touched. False error logs should be greatly reduced. 

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

No change

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

dcnielsen90 committed:
bravia-tv version bump

dcnielsen90 committed:
Fix Auth Cookies

    - Fix authentication cookie expiration.
    - Remove obsolete error handling (since bravia-tv==1.0.2)
    - Remove obsolete mac address references. (since bravia-tv==1.0.2)
    - Update tests.

bieniu commented on homeassistant/components/braviatv/media_player.py:
```diff
@@ -144,77 +144,87 @@ def __init__(self, client, name, pin, unique_id, device_info, ignored_sources):
+        if not playing_info:
+            self._channel_name = "App"
+            playing_info = {}
+        self._program_name = playing_info.get("programTitle")
```
I think that `else` is missing here.

dcnielsen90 replied:
That was intentional to set all the other values to ```None```; however you are correct about it being wrong.  It should be:
```python
    async def _async_refresh_playing_info(self):
        """Refresh Playing information."""
        playing_info = await self.hass.async_add_executor_job(
            self._braviarc.get_playing_info
        )
        self._program_name = playing_info.get("programTitle")
        self._channel_name = playing_info.get("title")
        self._program_media_type = playing_info.get("programMediaType")
        self._channel_number = playing_info.get("dispNum")
        self._content_uri = playing_info.get("uri")
        self._source = self._get_source()
        self._duration = playing_info.get("durationSec")
        self._start_date_time = playing_info.get("startDateTime")
        if  self._channel_number is None:
            self._channel_name = "App"
```
I'll wait to see if you agree before I update it.

bieniu replied:
When TV gets signal from external source (HDMI, AV) `dispNum` is `None`.
I think
```python
playing_info = await self.hass.async_add_executor_job(
    self._braviarc.get_playing_info
)
self._program_name = playing_info.get("programTitle")
self._channel_name = playing_info.get("title")
self._program_media_type = playing_info.get("programMediaType")
self._channel_number = playing_info.get("dispNum")
self._content_uri = playing_info.get("uri")
self._source = self._get_source()
self._duration = playing_info.get("durationSec")
self._start_date_time = playing_info.get("startDateTime")
if not playing_info:
    self._channel_name = "App"
```
is a better solution.

dcnielsen90 replied:
Ah, that was a typo (didn't mean channel_number) -- but regardless I agree. I'll update it.

dcnielsen90 committed:
fix async_refresh_playing_info

bieniu commented:
LGTM :+1: Great work! I think you should add yourself to the `CODEOWNERS` file.

MartinHjelmare commented on homeassistant/components/braviatv/media_player.py:
```diff
@@ -144,77 +144,86 @@ def __init__(self, client, name, pin, unique_id, device_info, ignored_sources):
-            _LOGGER.error(exception_instance)
-            self._state = STATE_OFF
+            self._state = STATE_ON
+            if await self._async_refresh_volume():
+                if await self._async_refresh_channels():
+                    await self._async_refresh_playing_info()
+                    return
```
```suggestion
            if (
                await self._async_refresh_volume()
                and await self._async_refresh_channels()
            ):
                await self._async_refresh_playing_info()
                return
```

MartinHjelmare commented on homeassistant/components/braviatv/media_player.py:
```diff
@@ -294,11 +303,15 @@ def set_volume_level(self, volume):
 
     def turn_on(self):
         """Turn the media player on."""
+        self._state_change = True
```
Why do we need this?

If we need to guard from parallel updates, set `PARALLEL_UPDATES` constant to 1 here in the media_player module.

dcnielsen90 replied:
I only wanted to protect against updating during weird states. Is ```PARALLEL_UPDATES```  a semaphore per TV or all TVs under this integration? ```turn_on()``` has a timeout of 10 seconds which it will commonly get close to if the TV is coming from a completely off state -- I wouldn't want to stop updates to other TVs because another TV is being turned on.

MartinHjelmare replied:
The `PARALLEL_UPDATES` semaphore is for all entities of the platform, ie all TVs for this integration.

If we want to guard the device interface just within this entity I think we would need an asyncio.Lock. It's not enough with just this boolean attribute `_state_change`, since the `turn_on` method will be run in the thread pool while `async_update` is run in the event loop, ie different threads.

dcnielsen90 replied:
Done. I hope it was acceptable that I choose not to attempt to acquire the lock during async_update(). Since bravia.turn_on() can take a bit, it would push async_update() into it's next update interval.

dcnielsen90 committed:
Config_entry needs mac address for HA restart

    This was mistakenly taken out. Home-Assistant needs to store mac address in
    case restarted. TV needs to be on to acquire mac address. This is neccesary
    to allow TV to be turned on by Home-Assistant after Home-Assistant is
    restarted.

dcnielsen90 committed:
Improve code format

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

dcnielsen90 committed:
Skip update() during state changes

    It is best for performance to skip updates during state change because
    bravia.turn_on() will commonly push async_update() beyond it's update
    interval.

MartinHjelmare reviewed (approved):
Looks good!

Just update the test with the mac address.

dcnielsen90 committed:
update tests

dcnielsen90 commented:
I think its all set. Thanks guys!
